BenchmarkTests/Runner/AppDelegate.swift

```diff
@@ -5,20 +5,76 @@
  */
 
 import UIKit
+import DatadogInternal
+import DatadogBenchmarks
 
 @main
 class AppDelegate: UIResponder, UIApplicationDelegate {
     var window: UIWindow?
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        let info = try! TestInfo() // crash if test info are missing or malformed
+        guard
+            let scenario = SyntheticScenario(),
+            let run = SyntheticRun()
+        else {
+            return false
+        }
 
-        let scenario: Scenario = SyntheticScenario() ?? DefaultScenario()
+        let applicationInfo = try! AppInfo() // crash if info are missing or malformed
+
+        switch run {
+        case .baseline, .metrics:
+            // measure metrics during baseline and metrics runs
+            Benchmarks.metrics(
+                with: Benchmarks.Configuration(
+                    info: applicationInfo,
+                    scenario: scenario,
+                    run: run
+                )
+            )
+        case .profiling:
```

Comment:
How will the `.profiling` use case differ from `.metrics` in practice?

Reply:
Yes, we will perform different runs from baseline, metrics, and profiling. This way we avoid skewing metrics data with profiling overhead.